src/ElmVersion.hs

```diff
@@ -58,3 +61,22 @@ style_0_19_cannotExposeOpenListing elmVersion =
         Elm_0_18 -> False
         Elm_0_18_Upgrade -> False
         _ -> True
+
+fromFile :: FileStore f => FilePath -> Free f ElmVersion
+fromFile path =
+    do
+        let dir = takeDirectory (path)
+        elmPackageJson <- stat (path </> "elm-package.json")
```

Comment:
Using `path` instead of `dir` allows to also work with directories like `"."`, the small drawback is an additional test for files. We could test if it's a directory instead, but this makes the logic more complicated. As you wish.

Reply:
Also `doesDirectoryExist` does not exist in `TestWorld.hs`, so this would require a little more work.